### PR TITLE
[StubV] Add support for ANDI Instruction

### PR DESCRIPTION
**Pull request by mmoadeli**
- Add support for ANDI Instruction
- PAT for AND and ANDI isel
- Test

### CODE REVIEW

File: llvm/lib/Target/StubV/StubVInstrInfo.td [hunk 1]
```diff
@@ -325,6 +325,39 @@ def AND : Instruction {
...
+  let Inst{19-15} = rs1;
+  let Inst{14-12} = 0b111;
+  let Inst{11-7} = rd;
+  let Inst{6-0} = OPC_OP_IMM.Value;
+  let isCommutable = 1;
+  let hasSideEffects = 0;
+  let mayLoad = 0;
+  let mayStore = 0;
+
+  let isReMaterializable = 1;
```
Comment (frasercrmck) [inline llvm/lib/Target/StubV/StubVInstrInfo.td]: I'm don't think `ANDI` is rematerializable or cheap-as-a-move.

File: llvm/test/CodeGen/StubV/and.ll [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+; RUN: llc -mtriple=stubv32 < %s | FileCheck %s
+
+target datalayout = "e-m:e-p:32:32-i64:64-n32-S128"
+target triple = "stubv32"
+
+define i32 @and(i32 %a, i32 %b) {
+; CHECK-LABEL: and:
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    and a0, a0, a1
+; CHECK-NEXT:    andi a0, a0, 42
```
Comment (frasercrmck) [inline llvm/test/CodeGen/StubV/and.ll]: You might want to add an alias from `andi` to `add`, as we've done with ADDI. The instruction is still ANDI - in the ISA and internally to the compiler - but the assemblers both accept the reg/reg form even with an immediate operand. Pretty strange! It should be pretty easy to see how it's done though.

**Review by frasercrmck — COMMENTED** [review]
Looks good, thanks! Just a few little things to finish off.

File: llvm/lib/Target/StubV/StubVInstrInfo.td (hunk 1 above)
Comment (mmoadeli) [reply llvm/lib/Target/StubV/StubVInstrInfo.td]: Thank you. That's right. In RISCV is not also set as `isReMaterializable`.

File: llvm/test/CodeGen/StubV/and.ll (hunk 2 above)
Comment (mmoadeli) [reply llvm/test/CodeGen/StubV/and.ll]: Thanks @frasercrmck, done.

**Review by frasercrmck — APPROVED** [review]
LGTM, nice one

**Pull request merged**